backend/src/ml_space_lambda/app_configuration/lambda_functions.py

```diff
@@ -50,6 +147,41 @@ def update_configuration(event, context):
             e.response["ResponseMetadata"]["HTTPStatusCode"] = 429
         raise e
 
+    # Create the new app deny policy
+    try:
+        deny_policy_statements = []
+        enabled_services_dict = app_configuration.configuration.enabled_services.to_dict()
+        # Check each service
+        for service in enabled_services_dict.keys():
```

Comment:
If there are no disabled services just bail here rather than trying to create an empty policy.

Reply:
This is now covered. In the case that there are no deactivated services a filler statement will be applied for scalability reasons.

In order to delete a policy [it must be detached](https://boto3.amazonaws.com/v1/documentation/api/1.28.1/reference/services/iam/client/delete_policy.html)

IAM is limited to [20 transactions per second](https://docs.aws.amazon.com/singlesignon/latest/userguide/limits.html#ssothrottlelimits) (TPS). So, one of two problems would occur when trying to delete. Either it would take a long time to detach the policy from all roles and then delete it, or the throttling limit would be hit and trigger a failure.

For this reason a filler statement will save us from either of these issues at the expense of having a policy with a meaningless statement. Seems like a small and valid price to pay to retain the attachments.